rasa_core/agent.py

```diff
@@ -419,6 +468,85 @@ def _create_ensemble(policies):
         else:
             passed_type = type(policies).__name__
             raise ValueError(
-                    "Invalid param `policies`. Passed object is "
-                    "of type '{}', but should be policy, an array of "
-                    "policies, or a policy ensemble".format(passed_type))
+                "Invalid param `policies`. Passed object is "
+                "of type '{}', but should be policy, an array of "
+                "policies, or a policy ensemble".format(passed_type))
+
+    @staticmethod
+    def _get_model_hash(model_server, header='model_hash'):
```

Comment:
can we use a standard header field for this, e.g. the `ETag` header?

Reply:
thinking about this more, we could even just use the `GET` request to retrieve the model, sending a header `If-None-Match: "PREVIOUS_MODEL_HASH"`, the standard says that the server should return `204 - Not modified` with an empty body if the resource has not changed and otherwise should return the resource.

This would move this checking logic to the backend and rasa core would also work with a model storage backend like s3